### PR TITLE
Sync 1.14 branch with develop

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,11 +20,11 @@ DOCUMENTATION
 -------------
 This release is fully functional for the API described in the documentation.
 
-   https://portal.hdfgroup.org/display/HDF5/The+HDF5+API
+   https://docs.hdfgroup.org/hdf5/develop/_l_b_a_p_i.html
```
Comment (byrnHDF) [inline README.md]: develop S.B. hdf5_1_14 - followup PR?

File: test/API/README.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ Currently unsupported
 
 These API tests currently only support usage with the native HDF5 VOL connector and HDF5 VOL
 connectors that can be loaded dynamically as a plugin. For information on how to build a VOL
-connector in this manner, refer to section 2.3 of the [HDF5 VOL Connector Author Guide](https://portal.hdfgroup.org/display/HDF5/HDF5+VOL+Connector+Authors+Guide?preview=/53610813/59903039/vol_connector_author_guide.pdf).
+connector in this manner, refer to section 2.3 of the [HDF5 VOL Connector Author Guide](https://docs.hdfgroup.org/hdf5/develop/_v_o_l__connector.html).
```
Comment (byrnHDF) [inline test/API/README.md]: develop -> hdf5_1_14

**Review by byrnHDF — COMMENTED** [review]
Other then the three develop branch to hdf5_1_14 branch changes - cmake, doxygenand examples changes are good.

File: README.md (hunk 1 above)
Comment (lrknox) [reply README.md]: Fixed

File: test/API/README.md (hunk 2 above)
Comment (lrknox) [reply test/API/README.md]: Fixed

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**